src/admin-page/UsersTable.js

```diff
@@ -0,0 +1,245 @@
+import React, { useState } from 'react';
+import {
+  Button,
+  IconButton, Input, Modal, Paper,
+  Table, TableBody, TableCell, TableContainer, TableHead, TableRow,
+} from '@material-ui/core';
+import { useMutation, useQuery } from '@apollo/client';
+import { PersonAdd, Done, Cancel } from '@material-ui/icons';
+import UserRow from './UserRow';
+import {
+  CREATE_USER, DELETE_USER, GET_USERS, UPDATE_USER,
+} from '../sources';
+import './styles.css';
+import UserModal from './UserModal';
+
+function getUserRoleId(userRoleName) {
+  if (userRoleName === 'sys_admin') return 1;
+  if (userRoleName === 'company_user') return 2;
+  if (userRoleName === 'user') return 3;
+  return null;
+}
+
+const UsersTable = () => {
+  const { data, error } = useQuery(GET_USERS);
+  const [deleteUserMutation] = useMutation(DELETE_USER);
+  const [updateUserMutation] = useMutation(UPDATE_USER);
+  const [createUser] = useMutation(CREATE_USER);
+
+  const [addRowButtonClicked, setAddRowButtonClicked] = useState(false);
+  const [openDeleteModal, setOpenDeleteModal] = useState(false);
+  const [openUpdateModal, setOpenUpdateModal] = useState(false);
+  const [idToDelete, setIdToDelete] = useState(null);
```

Comment:
Can you perform a delete and an update in the same time ? Otherwise storing just one id would be enough.

Reply:
I was thinking that i can do them at the same time so that way i will avoid errors for special use cases, but it's not the case. Fixed